test/test_color.rb

```diff
@@ -0,0 +1,61 @@
+require "minitest/autorun"
+require "gosu" unless defined? Gosu
+
+class TestConstants < Minitest::Test
+  def test_predefined_colors
+    [:NONE, :BLACK, :GRAY, :WHITE, :AQUA, :RED, :GREEN, :BLUE, :YELLOW, :FUCHSIA, :CYAN].each do |base_color|
+      assert Gosu::Color.const_defined?(base_color)
+    end
+  end
+
+  def test_color_creation
+    # ARGB
+    assert_equal Gosu::Color::RED,   Gosu::Color.new( 0xff, 0xff, 0x00, 0x00)
+    assert_equal Gosu::Color::GREEN, Gosu::Color.argb(0xff, 0x00, 0xff, 0x00)
+    assert_equal Gosu::Color::BLUE,  Gosu::Color.rgba(0x00, 0x00, 0xff, 0xff)
+
+    # (A)HSV
+    assert_equal Gosu::Color::FUCHSIA, Gosu::Color.from_hsv(     300, 1.0, 1.0)
+    assert_equal Gosu::Color::YELLOW,  Gosu::Color.from_ahsv(255, 60, 1.0, 1.0)
+  end
+
+  def test_color_atttributes_and_types
+    color = Gosu::Color.new(255,255,255,255)
+    [:alpha, :red, :green, :blue, :hue].each do |attr|
+      assert_respond_to color, attr
+      assert_kind_of Integer, color.send(attr), "Color##{attr} should be kind of Integer"
+    end
+
+    [:saturation, :value].each do |attr|
+      assert_respond_to color, attr
+      assert_kind_of Float, color.send(attr), "Color##{attr} should be kind of Float"
+    end
+  end
+
+  def test_color_attribute_ranges
+    # atm this automatically sets the values to max, min if the exceed the range ...
+    assert_equal Gosu::Color::WHITE, Gosu::Color.new(300, 300, 300, 300)
+    assert_equal Gosu::Color::NONE,  Gosu::Color.new(-50, -50, -50, -50)
+
+    # ... but this does not. I'm actually in favor of raising and exception if either
+    # of the creation methods get invalid values or at least let them behave them# same.
```

Comment:
The RGBA setters should not raise exceptions. Harmless mistakes shouldn't crash the game: `color.alpha = 128 + 128 * Math.sin(...)`.
Hue is treated like an angle, and is not clamped but %-ed into the [0; 360) range. I've pushed a short commit to clamp saturation and value. Thanks for pointing this out.